spec/support/expect_script_helper.rb

```diff
@@ -0,0 +1,50 @@
+class ScriptOutput < RSpec::Matchers::BuiltIn::Output
+  def matches?(block)
+    @block = block
+    return false unless Proc === block
+    @source = block.call
+    @actual = @stream_capturer.capture lambda { run_example(@source) }
+    @expected ? values_match?(@expected, @actual) : captured?
+  end
+
+  def failure_message
+    "expected:\n#{indented_source}\nto #{description}, but #{positive_failure_reason}"
+  end
+
+  def failure_message_when_negated
+    "expected:\n#{indented_source}\nto not #{description}, but #{negative_failure_reason}"
+  end
+
+  def indented_source
+    indent = @source.split("\n").select {|line| !line.strip.empty? }.map {|line| line.index(/[^\s]/) }.compact.min || 0
+    @source.gsub(/^[[:blank:]]{#{indent}}/, '  ')
+  end
+
+  def in_tmpdir(&block)
+    Dir.mktmpdir('bootscale') { |dir| Dir.chdir(dir, &block) }
+  end
+
+  def run_example(code)
+    in_tmpdir do
+      files = code.split(/^ +# (.*\.rb)\n/)[1..-1].each_slice(2).to_a
```

Comment:
not sure how this is working when the indentation is supposed to be removed since it matches on `^ +`